### PR TITLE
setup for cn/htf/snd flags

### PR DESCRIPTION
**Pull request by SidelineCory24**
Setup/CSS for each flag

- Critical Need is set to show on every position card (not hooked up yet to BE)

### CODE REVIEW

File: src/sass/_ribbon.scss [hunk 1]
```diff
@@ -38,6 +38,18 @@ $ribbon-shadow-color: rgba(112, 112, 112, .5);
...
+}
+
+.ribbon-htf {
+  background-color: $primary-alt-base;
+}
+
+.ribbon-snd {
+  background-color: $tertiary-success-lighter;
+}
+
```
Comment (SidelineCory24) [inline src/sass/_ribbon.scss:52]: For color/shade: SND matches the ticket (I think), CN/HTF don't

Comment (SidelineCory24) [reply src/sass/_ribbon.scss:52]: Elizabeth = 🐐, thanks for the help with me this!

File: src/Components/Ribbon/CriticalNeed/CriticalNeed.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import Ribbon from '../Ribbon';
+
+const CriticalNeed = ({ condensed, compare, ...props }) => {
+  let text = 'Critical need';
+  if (condensed) {
+    text = '';
+  }
+  if (compare) {
+    text = 'CN';
+  }
```
Comment (SidelineCory24) [inline src/Components/Ribbon/CriticalNeed/CriticalNeed.jsx:11]: I think there is a cleaner way to do this but I couldn't think of it. This is assuming we'll go with the abbreviated name for the compare list ribbon.

Comment (elizabeth-jimenez) [reply src/Components/Ribbon/CriticalNeed/CriticalNeed.jsx:11]: i would look into the `showText` property in the `Ribbon` component. If we can use that, then we end up just needing to handle 2 cases here: 'Critical need' and 'CN'

```
const text = condensed ? 'CN' : 'Critical need';
```

Comment (SidelineCory24) [reply src/Components/Ribbon/CriticalNeed/CriticalNeed.jsx]: done

File: src/sass/_condensedCard.scss [hunk 3]
```diff
@@ -10,7 +10,7 @@ $condensed-card-data-padding: 15px 20px 9px;
   font-size: 16px;
   line-height: 19px;
   margin-bottom: $condensed-card-margin-bottom;
-  width: 350px;
+  width: 360px;
```
Comment (SidelineCory24) [inline src/sass/_condensedCard.scss:13]: This width along with the width in `ribbon.scss` were changed for the edge case of the longest position title and skill on one line combined with all four flags showing. I tested making only the ribbon smaller, but it seemed too small to me.

These are the test I used:
title: INFO MANAGEMENT TECH SPEC/DIGI
skill: U.S. CITIZEN SERVICES (3055)

File: src/Components/Ribbon/ServiceNeedDifferential/ServiceNeedDifferential.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+const ServiceNeedDifferential = ({ condensed, compare, shortName, ...props }) => {
+  const text = shortName ? 'SND' : 'Service need differential';
+  // let text = 'Service need differential';
+  // if (condensed) {
+  //   text = '';
+  // }
+  // if (compare) {
+  //   text = 'SND';
+  // }
```
Comment (elizabeth-jimenez) [inline src/Components/Ribbon/ServiceNeedDifferential/ServiceNeedDifferential.jsx:12]: 🧼

File: src/Components/BureauPage/BureauResultsCard/BureauResultsCard.jsx [hunk 5]
```diff
@@ -86,7 +86,16 @@ class BureauResultsCard extends Component {
             <div>{postShort}</div>
             <div className="shortlist-icon">{shortListIndicator}</div>
             {
-              get(stats, 'has_handshake_offered', false) && <Handshake isWide className="ribbon-results-card" />
+              get(stats, 'has_handshake_offered', false) && <Handshake isWide cutSide="both" className="ribbon-results-card" />
+            }
+            { // need to verify if this is in the payload
```
Comment (elizabeth-jimenez) [inline src/Components/BureauPage/BureauResultsCard/BureauResultsCard.jsx:91]: 🧼

File: src/sass/_ribbon.scss [hunk 6]
```diff
@@ -73,12 +98,43 @@ $ribbon-shadow-color: rgba(112, 112, 112, .5);
...
+  position: absolute;
+  right: -1px;
+  top: calc(100% - 220px);
+  width: 170px;
+
+  .ribbon-hs {
+    max-width: 150px;
+    float: right;
+    padding-left: 4em;
+    padding-right: 3.35em;
```
Comment (elizabeth-jimenez) [inline src/sass/_ribbon.scss:111]: the reason you're having to do a `max-width` on here is because of the large amount of padding(4, 3.35). If we set this padding to be the same as ribbon-cn and ribbon-htf, we won't need the max-width. Once we do that, there will be a lot of code repetition here, so look into optimizing that:
```
.class1 {
styles
}
.class2 {
styles
}
.class3 {
styles
}
```
can become something like:
```
.class1, .class2, .class3 {
styles
}
```

Comment (elizabeth-jimenez) [reply src/sass/_ribbon.scss:111]: i'd also look into `.ribbon-primary`; if we only use that class for the HS ribbon, then we can modify that class instead of creating `.ribbon-hs`(basically just move your styles for ribbon-hs to another place), that way we don't deprecate `.ribbon-primary`

File: src/Components/Ribbon/ServiceNeedDifferential/ServiceNeedDifferential.jsx (hunk 4 above)
Comment (SidelineCory24) [reply src/Components/Ribbon/ServiceNeedDifferential/ServiceNeedDifferential.jsx:12]: done

File: src/sass/_ribbon.scss (hunk 6 above)
Comment (SidelineCory24) [reply src/sass/_ribbon.scss:111]: done

**Comment by elizabeth-jimenez** [comment]
We can wrap the Critical Need stuff in `<StaticDevContent>` until we figure out what to map it to. Once we do know, we can just remove it, as part of the mapping update.

File: src/sass/_variables.scss [hunk 7]
```diff
@@ -118,6 +118,9 @@ $tm-gray: #AAAAAA;
 $tm-navy: #215493;
 $tm-navy-dark: $blue-primary-darkest;
 
+$cn-flag: #EDA176 ;
+$htf-flag: #89C1F2;
+
```
Comment (elizabeth-jimenez) [inline src/sass/_variables.scss:123]: I'd add a comment here, like 

//**
//* Flags (or Ribbons, whichever you think is best)
//*

and maybe flip it so that it's `flag-cn` and `flag-htf`; it looks like that's how we do it everywhere else in this file.

File: src/sass/_ribbon.scss [hunk 8]
```diff
@@ -73,12 +94,30 @@ $ribbon-shadow-color: rgba(112, 112, 112, .5);
   right: -1px;
 }
 
+.ribbon-container-results {
+  position: absolute;
+  right: -1px;
+  top: calc(100% - 220px);
+  width: 170px;
+
+  .ribbon-primary, .ribbon-cn, .ribbon-htf {
```
Comment (elizabeth-jimenez) [inline src/sass/_ribbon.scss:103]: 😍

File: src/Components/ResultsCondensedCardTop/ResultsCondensedCardTop.jsx [hunk 9]
```diff
@@ -27,6 +27,9 @@ const ResultsCondensedCardTop = ({
   const p = position.position || position;
   const stats = getBidStatisticsObject(position.bid_statistics);
   const hasHandshake = get(stats, 'has_handshake_offered', false);
+  // const isCriticalNeed = ???
```
Comment (elizabeth-jimenez) [inline src/Components/ResultsCondensedCardTop/ResultsCondensedCardTop.jsx:30]: 🧼

File: src/Components/ResultsCard/ResultsCard.jsx [hunk 10]
```diff
@@ -300,9 +300,18 @@ class ResultsCard extends Component {
                   </Column>
                 }
                 <Column columns="2">
-                  <div className="ribbon-container">
+                  <div className="ribbon-container-results">
```
Comment (elizabeth-jimenez) [inline src/Components/ResultsCard/ResultsCard.jsx:303]: same comment as `.ribbon-primary`. It looks like this change has deprecated our `ribbon-container` class? Seems this is the only place this class was being used, so we can just directly modify `ribbon-container` in src/sass/_ribbon.scss, instead of creating a new class (same thing as ribbon-primary, just move the work you've already done for ribbon-container-results to ribbon-container)

File: src/Components/ResultsCondensedCardTop/ResultsCondensedCardTop.jsx [hunk 11]
```diff
@@ -60,9 +63,18 @@ const ResultsCondensedCardTop = ({
             }
           </span></span>
         </div>
-        <div className="ribbon-container">
+        <div className="ribbon-container-condensed">
```
Comment (elizabeth-jimenez) [inline src/Components/ResultsCondensedCardTop/ResultsCondensedCardTop.jsx:66]: 👍 this one should be alright, because we do want different styles from `ribbon-container`

File: src/Components/Ribbon/index.js [hunk 12]
```diff
@@ -1,7 +1,10 @@
 import Featured from './Featured';
 import Handshake from './Handshake';
+import CriticalNeed from './CriticalNeed';
+import HardToFill from './HardToFill';
+import ServiceNeedDifferential from './ServiceNeedDifferential';
 import InBidList from './InBidList';
 
-export { Handshake, Featured, InBidList };
+export { Handshake, CriticalNeed, HardToFill, ServiceNeedDifferential, Featured, InBidList };
```
Comment (elizabeth-jimenez) [inline src/Components/Ribbon/index.js:8]: 🎊  these files are easy to forget, so good job!

File: src/Components/Ribbon/Ribbon.jsx [hunk 13]
```diff
@@ -22,7 +22,7 @@ Ribbon.propTypes = {
   icon: PropTypes.string,
   text: PropTypes.string,
   containerProps: PropTypes.shape({}),
```
Comment (elizabeth-jimenez) [inline src/Components/Ribbon/Ribbon.jsx:24]: line 6 🔍

Comment (elizabeth-jimenez) [reply src/Components/Ribbon/Ribbon.jsx:24]: see linked [comment](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/1431/files#r613565191)

File: src/sass/_variables.scss (hunk 7 above)
Comment (SidelineCory24) [reply src/sass/_variables.scss:123]: done

File: src/Components/ResultsCondensedCardTop/ResultsCondensedCardTop.jsx (hunk 9 above)
Comment (SidelineCory24) [reply src/Components/ResultsCondensedCardTop/ResultsCondensedCardTop.jsx:30]: done

File: src/Components/ResultsCard/ResultsCard.jsx (hunk 10 above)
Comment (SidelineCory24) [reply src/Components/ResultsCard/ResultsCard.jsx:303]: done

File: src/Components/Ribbon/Ribbon.jsx (hunk 13 above)
Comment (SidelineCory24) [reply src/Components/Ribbon/Ribbon.jsx:24]: @elizabeth-jimenez this brings me back to the files changed page, not sure what I should be looking at 😅. Do I need to make changes to className?

Comment (elizabeth-jimenez) [reply src/Components/Ribbon/Ribbon.jsx:24]: 😅  Line 6 in this file uses the `ribbon-outer-container` class. I think you removed the only styling for this class in file `src/sass/_details.scss`. If so, using that class now does nothing, so we are safe to go ahead and remove it

File: src/Components/BureauPage/BureauResultsCard/BureauResultsCard.jsx [hunk 14]
```diff
@@ -88,8 +89,10 @@ class BureauResultsCard extends Component {
             {
               get(stats, 'has_handshake_offered', false) && <Handshake isWide cutSide="both" className="ribbon-results-card" />
             }
-            { // need to verify if this is in the payload
-              <CriticalNeed isWide cutSide="both" className="ribbon-results-card" />
+            {
+              <StaticDevContent>
```
Comment (elizabeth-jimenez) [inline src/Components/BureauPage/BureauResultsCard/BureauResultsCard.jsx:93]: 🦜

File: src/sass/_variables.scss [hunk 15]
```diff
@@ -118,8 +118,10 @@ $tm-gray: #AAAAAA;
 $tm-navy: #215493;
 $tm-navy-dark: $blue-primary-darkest;
 
-$cn-flag: #EDA176 ;
-$htf-flag: #89C1F2;
+//**
+//* Ribbon Colors
+$ribbon-cn: #EDA176 ;
+$ribbon-htf: #89C1F2;
```
Comment (elizabeth-jimenez) [inline src/sass/_variables.scss:124]: 😍

File: src/Components/Ribbon/Ribbon.jsx (hunk 13 above)
Comment (SidelineCory24) [reply src/Components/Ribbon/Ribbon.jsx:24]: Ah okay I got it  🤦🏾‍♂️

Comment (SidelineCory24) [reply src/Components/Ribbon/Ribbon.jsx:24]: done

**Comment by elizabeth-jimenez** [comment]
image
Still seeing the In BidList error

**Comment by elizabeth-jimenez** [comment]
image
I feel like this wasn't like this before 🤔

**Comment by elizabeth-jimenez** [comment]
> image
> I feel like this wasn't like this before 🤔

image
looks like a styling on HS is what moves CN over

**Comment by elizabeth-jimenez** [comment]
image
i think the solution here is to keep the box shadow stuff how it was and add the `clip-path: polygon(9% 0, 100% 0, 91% 100%, 0 100%, 0 100%);`

**Comment by SidelineCory24** [comment]
Looks like the `StaticDevContent` is causing the issue on the Bureau results card. Should we remove it for this file?

**Comment by SidelineCory24** [comment]
Bureau Position Details
image

**Comment by SidelineCory24** [comment]
Bureau Results Card, couldn't find a good way to handle the box shadow.
image

**Comment by SidelineCory24** [comment]
Compare List
image

**Comment by SidelineCory24** [comment]
Bidder/CDO Position Details
image

**Comment by SidelineCory24** [comment]
Search Results 
image

**Comment by SidelineCory24** [comment]
Condensed Card
image

**Comment by SidelineCory24** [comment]
These are the designs I'll show to Chris and Sarah.

**Review by elizabeth-jimenez — APPROVED** [review]
🍰 🧁 🍬 👍  cool job!
